core/embed/rust/src/ui/component/base.rs

```diff
@@ -208,28 +208,44 @@ where
 /// Same as `Child` but also handles screen clearing when layout is first
 /// painted.
 pub struct Root<T> {
-    inner: Child<T>,
+    inner: Option<Child<T>>,
     marked_for_clear: bool,
 }
 
 impl<T> Root<T> {
     pub fn new(component: T) -> Self {
         Self {
-            inner: Child::new(component),
+            inner: Some(Child::new(component)),
             marked_for_clear: true,
         }
     }
 
     pub fn inner(&self) -> &Child<T> {
-        &self.inner
+        if let Some(ref c) = self.inner {
+            c
+        } else {
+            fatal_error!("deallocated", "Root object is deallocated")
+        }
     }
 
     pub fn skip_paint(&mut self) {
-        self.inner.skip_paint()
+        if let Some(ref mut c) = self.inner {
+            c.skip_paint();
+        } else {
+            fatal_error!("deallocated", "Root object is deallocated")
+        }
```

Comment:
extract this into `pub fn inner_mut(&mut self)` and then call `self.inner_mut().skip_paint()` etc. in the other functions

Reply:
fixed https://github.com/trezor/trezor-firmware/commit/ee7f88dddc0c6a9e9708e8a30c045a4e411ce7fe